e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -110,32 +124,34 @@ func NewPortForwarder(
 func (pf *PortForwarder) LocalPort() (int, error) {
 	pf.mux.Lock()
 	defer pf.mux.Unlock()
-	if pf.localPort == 0 {
-		err := pf.start()
-		return pf.localPort, err
+	if !pf.isReady {
+		return pf.localPort, fmt.Errorf("PortForwarder for Deployment %s/%s is not ready", pf.ns, pf.deployment)
 	}
 	return pf.localPort, nil
 }
 
-func (pf *PortForwarder) setPort(cmd *exec.Cmd, port int, pod string, async bool) {
+func (pf *PortForwarder) setPort(cmd *exec.Cmd, port int, pod string) {
 	pf.logger.Infof("updating port-forward %s:%d -> %s:%d", pf.pod, pf.localPort, pod, port)
-	if async {
-		pf.mux.Lock()
-	}
+	pf.mux.Lock()
 	pf.cmd = cmd
 	pf.localPort = port
 	pf.pod = pod
-	if async {
-		pf.mux.Unlock()
-	}
+	pf.isReady = true
+	pf.mux.Unlock()
 	if pf.setPortCallback != nil {
 		pf.setPortCallback(port, pod)
 	}
 }
 
+func (pf *PortForwarder) notReady() {
```

Comment:
I think you can just re-use the above `update` func and pass a ready arg.

Reply:
We don't want to invoke the callback here, or necessarily need to set any of the other parameters. I think we would be bloating the `update` function if we did this

Reply:
You can just wrap the callback in `if ready`, but :shrug: